### ISSUE TITLE
"lógica de programação" é um termo que NÃO existe

### ISSUE THREAD

**Issue by terremoth** [pilhacheia/tretadev#65]
Treta iniciada primeiro pelo Paulo Torrens: https://www.facebook.com/groups/142918099147059/posts/488111991294333/
Redigida pro github caso ele apagasse de lá: https://gist.github.com/vinicius73/8d121301de6788528bf8a55e824caa97

E por fim, uma palestra que ele deu sobre isso na UDESC: https://www.youtube.com/watch?v=VpzhH9CXOy0&ab_channel=Col%C3%B3quiosdeCi%C3%AAnciadaComputa%C3%A7%C3%A3o

**Comment by Joshaby** [comment]
Paulo utorrent é conhecido até aqui

**Comment by lincolixavier** [comment]
Gravamos podcast com o Torrens, vale ouvir xD


https://open.spotify.com/show/2bBM3Lv7k3tEtff6S8MIGK?si=AYGc9xiYRBeP4Fu2F77DgQ&dl_branch=1

**Comment by adolfont** [comment]
Existir existe, mas não é a mesma coisa que lógoca matemática/filisófica.

**Comment by terremoth** [comment]
Não, não existe, meu chute é que esse termo foi inventado por aqui pelo brasil mesmo.
Não existe formalidade de "programming logic". (o que existe é logic programming que é um paradigma de programação que não tem nada a ver com o assunto).

O que as pessoas usam como "lógica de programação" é na verdade "estruturas semânticas e sintáticas de linguagens imperativas".

**Comment by adolfont** [comment]
> Não, não existe, meu chute é que esse termo foi inventado por aqui pelo brasil mesmo nos anos 80. Não existe formalidade de "programming logic". (o que existe é "logic programming" que é um paradigma de programação que não tem nada a ver com o assunto).

Oi, @terremoth. Existe sim. Não foi inventado no Brasil, acho.

Exemplos de textos em sites não-brasileiros:

"What Does Programming Logic Mean?
Programming logic is a fundamental construct that's applied to computer science in a variety of comprehensive ways.

Programming logic involves logical operations on hard data that works according to logical principles and quantifiable results."  https://www.techopedia.com/definition/3945/programming-logic

"How to Improve Your Programming Logic" https://woz-u.com/blog/how-to-get-programming-logic/

Programação em Lógica, ou Programação no Paradigma Lógico (_Logic Programming_) é outra coisa de fato.

O termo de fato saiu de moda. Usam mais "Computational Thinking" (que não é a mesma coisa mas serve a um propósitivo parecido). Mas você ainda encontra em universidades de fora (menores) cursos de Programming Logic: https://www.trinity.edu/academics/cosb/csci/introduction-programming-logic

**Comment by adolfont** [comment]
Mais: neste artigo de 1974 você encontra o termo "programming logic" https://www.tandfonline.com/doi/abs/10.1080/00220310.1974.11646172?journalCode=ucis19

**Comment by terremoth** [comment]
Techopedia > "April 30, 2020"
Woz-u > "June 15th, 2020"

Se vc procurar talvez vc encontre outros lugares "escritos em inglês" o termo "programming logic" (é só pôr entre aspas duplas na pesquisa do Google). Tem um livro escrito por uma mulher também, se não me engano, que leva o titulo ou subtitulo de "programming logic". 

E tá claro que tu procurou isso aí no google 3h atrás pra responder kkkk
Esse termo não é comum e formal, é inventado.

É até sem lógica (pun-intended, ba-dum-tsss) se for pensar no termo "lógica de programação", pois nesse termo implica que existe algum tipo de lógica _dentro_ do assunto programação, como se lógica fosse um subconjunto de programação. E isso é mentira, o que existem são tipos de lógica (que tem origem na [filosofia [isso mesmo que você leu]](https://pt.wikipedia.org/wiki/L%C3%B3gica)) como por exemplo lógica modal, lógica de predicados, lógica difusa, etc que SE APLICA/SE CODIFICA dentro dos algoritmos e nem todas as linguagens de programação dão suporte a "IF's, OR, AND, NOT, etc". Portanto é simplesmente um misnomer usar esse termo para querer explicar "uso de ifs/else/for/while/and/or/not" em programação. É muito mais claro e fácil chamar de "computational thinking" como colocaste ou simplesmente "programação imperativa" ou "algoritmos imperativos" que é mais correto ao meu ver.

**Comment by adolfont** [comment]
> E tá claro que tu procurou isso aí no google 3h atrás pra responder kkkk Esse termo não é comum e formal, é inventado.

Eu procurei quando vi sua dúvida. Não lembrava de ter visto "programming logic" mas nunca pensei que tivesse sido inventado no Brasil. E de fato não foi inventado no Brasil.

**Comment by adolfont** [comment]
>E isso é mentira, o que existem são tipos de lógica (que tem origem na [filosofia [isso mesmo que você leu]](https://pt.wikipedia.org/wiki/L%C3%B3gica)) como por exemplo lógica modal, lógica de predicados, lógica difusa, etc que SE APLICA/SE CODIFICA dentro dos algoritmos e nem todas as linguagens de programação dão suporte a "IF's, OR, AND, NOT, etc". Portanto é simplesmente um misnomer usar esse termo para querer explicar "uso de ifs/else/for/while/and/or/not" em programação. É muito mais claro e fácil chamar de "computational thinking" como colocaste ou simplesmente "programação imperativa" ou "algoritmos imperativos" que é mais correto ao meu ver.


Definitivamente Lógica de programação não é uma lógica. Você está certo e eu nunca disse que Lógica de Programação era uma Lógica.

**Comment by gwitus** [comment]
Pior que lógica de programação é só o 'programação é que nem receita de bolo'
